### DISCUSSION TITLE
Message delete

### DISCUSSION THREAD

**Discussion by Jexly0x** [telegraf/telegraf#1626] [Q&A] [answered]
Hello. I have a simple telegram bot. There is a group to which I write messages, and the bot takes these messages from the channel and copies them into itself. If I delete a message in the channel, the bot does not repeat this in itself. How can I implement it?

![photo](https://user-images.githubusercontent.com/97974591/169719286-5cc56f81-639c-4b6c-be69-d59cb6b423f5.png)

**Comment by MKRhere** [comment, marked as answer]
Bots can't see message deletion. Perhaps as a workaround, you can reply to the message you want to delete with `/delete`, and have the bot catch the update and delete the replied message and the `/delete` message. This way the bot knows which message to delete.

Now, propagating it to the other chat is a bit more involved. You'll want to store a mapping of `channelMessage -> chatMessage` in your database and use that to delete the `chatMessage` when the bot receives `/delete` in reply to a `channelMessage`.

**Reply by Jexly0x** [reply]
How do I know if a channel has deleted a message? How to tell the bot about it? How to delete in the bot itself? This requires action, which is not.

**Reply by MKRhere** [reply]
```TS
bot.on("channel_post", async ctx => {
	// check if this is channelPost is a text message
	if ("text" in ctx.channelPost) {
		// check if the message is just "/delete"
		if (ctx.channelPost.text.trim() === "/delete") {
			// delete the "/delete" message
			await ctx.deleteMessage();

			// check if it's replying to a channel post
			if (ctx.channelPost.reply_to_message) {
				const channelMsgId = ctx.channelPost.reply_to_message.message_id
				// delete the channel post it's replying to
				await ctx.deleteMessage(channelMsgId);

				const chatMsg = await getLinkedChatMsgFromDBSomehow(channelMsgId);
				// delete the linked chat message
				await ctx.deleteMessage(chatMsg.message_id);
			}

			// nothing more to do here, return out of this function
			return;
		}

		// continue your normal flow
	}
});
```

Basically don't delete the channel post manually, but reply to the post to delete with `"/delete"` and the bot will delete it on both sides.

**Reply by Jexly0x** [reply]
Thank you. I'll check it out soon and give you feedback! Maybe you can also explain with my other topic "reply"?
